src/main/kotlin/io/github/mojira/arisa/modules/RelatedGroupModule.kt

```diff
@@ -0,0 +1,76 @@
+package io.github.mojira.arisa.modules
+
+import arrow.core.Either
+import arrow.core.extensions.fx
+import arrow.core.left
+import arrow.core.right
+import arrow.syntax.function.partially1
+import io.github.mojira.arisa.domain.Comment
+import io.github.mojira.arisa.domain.Issue
+import io.github.mojira.arisa.modules.relatedgroup.AddTicketsRelatedGroupSubmodule
+import io.github.mojira.arisa.modules.relatedgroup.RelatedGroupSubmodule
+import java.time.Instant
+
+class RelatedGroupModule(
+    private val arisaUsername: String
+) : Module {
+    override fun invoke(issue: Issue, lastRun: Instant): Either<ModuleError, ModuleResponse> = with(issue) {
+        Either.fx {
+            val arisaComments = comments
+                .filter(::isStaffRestricted)
+                .filter(::userIsArisa)
+                .filter(::isProbablyARelatedGroup)
+            assertNotEmpty(arisaComments).bind()
+
+            val isError = arisaComments.any {
+                !it.body.isNullOrBlank()
```

Comment:
there is a isNotNull 👀

Reply:
no? there's only isNotBlank and isNotEmpty, neither of which check for null (based on syntax error)

Reply:
and regardless, I need both checks, and they are conveniently combined into one

Reply:
or actually no, don't need both - assert works with empty or null result, and I just need to pass the syntax

Reply:
```suggestion
                it.body != null
```
I guess this what you were thinking about?

Reply:
nvm, brainfart :P